posthog/hogql_queries/lifecycle_query_runner.py

```diff
@@ -85,9 +85,12 @@ def to_persons_query(self) -> str:
             )
 
     def run(self) -> LifecycleQueryResponse:
+        to_query = self.to_query()
+        query = to_query[0] if isinstance(to_query, list) else to_query
```

Comment:
Is this needed? Do the inherited class methods have to have the same return type as their base?

Reply:
I think you can override the return type in `TrendsQuery` and `TrendsQueryRunner` to whatever makes sense there. Not sure right now...